### ISSUE TITLE
Renders static pages with 'loading' state when using Redux to populate content

### ISSUE THREAD

**Issue by rumbogs** [stereobooster/react-snap#193]
I cannot provide a reproducible example as it will need a working redux app that calls a backend as well.

I have a CRA that is using Redux to fetch content from a CMS and build the page. In this case, `react-snap` should wait for the content and take the snapshot after the redux store updates.

I've added `window.__PRELOADED_STATE__` from the documentation but the contents of the pages is still made up of the loading state. In any case, the `preloadedState` object can be found in a string tag in the page though. I'm not sure if I implemented it correctly or if the documentation has omitted some steps.

**Comment by stereobooster** [comment]
I can not debug the issue based on this report. You can add log statements to fetch request to make sure data is fetched during the react-snap processing.

**Comment by rumbogs** [comment]
I'm not sure what's the general practice for handling redux and fetching data. Before using redux, the pages worked as intended, but after, they're empty. Does the documentation explain everything that's needed to make it work?

**Comment by stereobooster** [comment]
> Does the documentation explain everything that's needed to make it work?

It should. Though not sure how is it easy for newbie.

Can you construct small reproducible example? Otherwise it is really hard to help

**Comment by rumbogs** [comment]
I have the `Home` component at root level that fetches its copy from a CMS. The data is received and can be found in `window.__PRELOADED_STATE__` as expected.

I'm also using `react-router` which renders `Home` with `<Route path="/" exact component={Home} />`.

```
class Home extends React.Component {
  componentWillMount() {
    const { home, fetchHome } = this.props;
    if (!home) {
      console.log('home will mount, fetching data');
      fetchHome();
    }
  }

  render() {
    const { home } = this.props;
    console.log('home: ', home);

    return home ? (
      <Wrapper>
        <div>{home.title}</div>
        <div>{home.content}</div>
      </Wrapper>
    ) : (
      <div>Loading</div>
    );
  }
}
```

`fetchHome` looks like this:

```
const fetchHome = () => ({
  type: GET_HOME,
  async payload() {
    const data = await fetchHomeService();
    console.log('data: ', data);
    return data;
  },
});
```

The backend works correctly as I'm able to find the needed data in `window.__PRELOADED_STATE__`.

`react-snap` logs give me:
```
✏️  / log: home will mount, fetching data
✏️  / log: home:
✏️  / log: data: <getting correct data here>
```
The problem I see is `react-snap` mounts and renders the component __before__ the data is received, and never updates it. Is there a way to trigger an update after the action is fulfilled?

**Comment by stereobooster** [comment]
This is not how you suppose to use Redux:

```
async payload() {
    const data = await fetchHomeService();
    console.log('data: ', data);
    return data;
  },
```

unless there is some kind of middleware which handles promises for you. You can stub out CMS with static JSON files and cook full reproducible example, if you believe this is issue with react-snap and not an issue with how you use Redux.

You can read more about Redux here https://redux.js.org/advanced/async-actions or ask questions in their discord channel.

**Comment by rumbogs** [comment]
I'm using `redux-promise-middleware` for promises, that is not the issue, just putting it here as an example. The data I get is fine and is exactly the live one. The problem is that the component is not updating after I get the data, only __PRELOADED_STATE__ gets updated. Is there a way to get the component in `react-snap` to use the new data after fetching resolves?

I can't create a fully reproducible example right now, but I think I covered everything in the information I gave in the previous post. It's a pretty straighforward use case IMO.

**Comment by stereobooster** [comment]
> I can't create a fully reproducible example right now, but I think I covered everything in the information I gave in the previous post. It's a pretty straighforward use case IMO.

I mean there are two options here: either you or I need to create reproducible demo, otherwise it is hard to debug it. I guess, if you want this to be solved you need to create reproducible example. If you do not want to invest time into creating reproducible example, why do you expect I will invest time into fixing it.

**Comment by rumbogs** [comment]
Sorry, I wasn't trying to be pushy, the app is too complex for me to reproduce it right away and was only asking specific questions that might have pointed me in the right direction. 

Anyway, I've found out what the issues were. `async / await` wasn't playing nicely, I don't really get why, got an error about not using absolute paths for fetch. Second, there was a data mapping issue on our end. And thirdly we had to add an `include` property in package.json to get react-snap go through some dynamic routes we had in place.

So apart from the `async / await` thing, it seems to be working nicely. Thank you for your help and this library!
